src/envoy/http/service_control/handler_impl_test.cc

```diff
@@ -1063,6 +1065,80 @@ TEST_F(HandlerTest, HandlerReportWithoutCheck) {
   handler.callReport(&headers, &response_headers, &resp_trailer_);
 }
 
+class HandlerReportStatusTest : public HandlerTest {
+ protected:
+  void runTest(unsigned int http_response_code,
+               TestResponseHeaderMapImpl response_headers,
+               TestResponseTrailerMapImpl response_trailers,
+               Code expected_grpc_status) {
+    EXPECT_CALL(mock_stream_info_, responseCode())
+        .WillRepeatedly(Return(http_response_code));
+
+    setPerRouteOperation("get_header_key");
+    TestRequestHeaderMapImpl headers{
+        {":method", "GET"}, {":path", "/echo"}, {"x-api-key", "foobar"}};
+    CheckDoneFunc stored_on_done;
+    CheckResponseInfo response_info;
+    ServiceControlHandlerImpl handler(headers, mock_stream_info_, "test-uuid",
+                                      *cfg_parser_, test_time_, stats_);
+
+    ReportRequestInfo expected_report_info;
+    initExpectedReportInfo(expected_report_info);
+    expected_report_info.api_key = "foobar";
+    expected_report_info.status = Status::OK;
+    expected_report_info.http_response_code = http_response_code;
+    expected_report_info.grpc_response_code = expected_grpc_status;
+    EXPECT_CALL(*mock_call_, callReport(MatchesReportInfo(
+                                 expected_report_info, headers,
+                                 response_headers, response_trailers)));
+    handler.callReport(&headers, &response_headers, &response_trailers);
+  }
+};
+
+TEST_F(HandlerReportStatusTest, GrpcStatusFromTrailers) {
+  // Test: When grpc status trailer is present, it's value will be used in
+  // addition to HTTP status.
+  runTest(200, {{"content-type", "application/grpc"}}, {{"grpc-status", "7"}},
```

Comment:
I did not see tests for "trailers-only" where grpc-status is in the response headers.

Reply:
Oops, I mixed up the braces in a test. Fixed.